connector_salesforce/salesforce_backend/backend_model.py

```diff
@@ -0,0 +1,407 @@
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from __future__ import absolute_import
+import simplejson
+from ..lib.oauth2_utils import SalesforceOauth2MAnager
+from openerp import models, fields, api, exceptions
+from openerp.tools.translate import _
+from openerp.addons.connector import session as csession, connector
+from ..unit.importer_synchronizer import batch_import, delayed_batch_import
+from ..unit.exporter_synchronizer import batch_export, delayed_batch_export
+
+
+class SalesforceBackend(models.Model):
+    """Salesforce backend
+
+    Please refer to connector backend documentation
+
+    There are 2 supported ways to access a Salesforce instance
+    Oauth2 flow
+    -----------
+    In order to use it you have to add a remote application in Salesforce
+    and enable Oauth login.
+
+    The created remote access app must have following parameters:
+
+    Permitted Users -->	All users may self-authorize
+    Callback URL --> public_odoo_url/salesforce/oauth
+
+
+    Once done you have to manage your app ensure the
+    `Refresh token is valid until revoked` parameter is set.
+
+    The following authentication method are still exprimental
+
+    User Password flow
+    ------------------
+    This flow allows a user to connect to api using SOAP access
+    in order to get a token. This approach is simpler but less secure
+    The first step is to pass the domain of your Salesforce instance
+    and an access token straight in Odoo backend.
+
+    You must have the full URL e.g (https://na1.salesforce.com)
+    of your instance.
+
+    There are also two means of authentication:
+    - Using username, password and security token
+    - Using IP filtering, username, password and organizationId
+    """
+
+    _name = "connector.salesforce.backend"
+    _inherit = "connector.backend"
+    _description = """Salesforce Backend"""
+    _backend_type = "salesforce"
+
+    @api.model
+    def _select_versions(self):
+        """ Available versions
+
+        Can be inherited to add custom versions.
+
+        :return: list of tuple of available versions
+        :rtype: list
+        """
+        return self._select_versions_hook()
+
+    @api.model
+    def _select_versions_hook(self):
+        """ Available versions
+
+        Can be inherited to add custom versions.
+
+        :return: list of tuple of available versions
+        :rtype: list
+        """
+        return [('15', "Winter'15")]
+
+    authentication_method = fields.Selection(
+        [
+            ('pwd_token', 'Based on User, Password, Token'),
+            ('oauth2', 'OAuth 2'),
+            ('ip_filtering', 'Based on IP Filter and OrganizationId')
+        ],
+        string='Authentication Method',
+        default='oauth2',
+    )
+
+    name = fields.Char(
+        'Name',
```

Comment:
You don't need this string. By default string is equal ta Capitalized field name.